thavalon-webapp/src/components/formComponents/InputElement.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import "../../styles/formStyles/InputElement.scss";
+
+export interface InputElementProps {
+    "type": string,
+    "label": string,
+    "name": string,
+    "required": boolean,
+    "minLength": number,
+    "formRef": any, // not fully clear what the type of register from react-hook-form is
+}
+
+/**
+ * A input component, with appropriate styling.
+ * @param props The props for the input element.
+ */
+export function InputElement(props: InputElementProps) {
```

Comment:
This function returns a JSX.Element, but that is missing from the return type hint.